File: BackEnd/proyecto/src/main/java/com/example/proyecto/model/Horario.java
package com.example.proyecto.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "horario")
public class Horario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "id_asignatura")
    private Asignatura asignatura;

    @Enumerated(EnumType.STRING)
    @Column(name = "dia")
    private Dia dia;

    @Column(name = "hora", nullable = false)
    private Integer hora;

    @Column(name = "id_curso")
    private Integer idCurso;

    @Column(name = "aula")
    private Integer aula;

    @Column(name = "id_profesor")
    private Integer idProfesor;
}

File: FrontEnd/app/src/main/java/com/example/tycep_fe/models/Horario.kt
package com.example.tycep_fe.models

import com.google.gson.annotations.SerializedName

data class Horario(
    @SerializedName("id")
    val id:Int,
    @SerializedName("asignatura")
    val asignatura:Asignatura,
    @SerializedName("dia")
    val dia:Dia,
    @SerializedName("hora")
    val hora:Int,
    @SerializedName("aula")
    val aula:Int,
    @SerializedName("id_profesor")
    val idProfesor: Int
)
